### PR TITLE
Add negatable Boolean flag primitive

### LINKED ISSUE #801: Support merging flags into a single help line
Right now I have something like this in a parser:
```ts
  color: optional(
    or(
      flag(`--color`, {
        description: message`Force-enable or force-disable ANSI syntax highlighting (default: auto, on when stdout is a TTY; honors NO_COLOR and FORCE_COLOR)`,
      }),
      map(flag(`--no-color`), () => false as const),
    ),
  ),
```

Which works how you'd expect, but then the generated help has two lines, one for `--color` and one for `--no-color`.

Ideally I could make it format like this:
```
--color, --no-color      Force-enable or force-disable ANSI syntax highlighting (default: auto, on when stdout is a TTY; honors NO_COLOR and FORCE_COLOR)
```

### PR DESCRIPTION
**Pull request by dahlia**
This adds `negatableFlag()` to *@optique/core* for paired Boolean flags such as `--color` and `--no-color`.

The new primitive returns `true` for positive names and `false` for negative names. When neither side is provided, it fails like `flag()`, so callers can choose their own absence behavior with `optional()` or `withDefault()`. This keeps runtime defaults explicit while still letting users override a setting in either direction.

The implementation in *packages/core/src/primitives.ts* supports aliases on both sides, one-line help entries, shell suggestions, hidden visibility modes, duplicate and conflict diagnostics, joined-value rejection, and custom error messages. Tests in *packages/core/src/primitives.test.ts* cover the main parser behavior, edge cases, and fast-check properties. *packages/core/src/public-api.test.ts* verifies the root export.

The documentation in *docs/concepts/primitives.md* and *docs/cookbook.md* describes the new API and shows how to combine it with `optional()` and `withDefault()`. The runnable pattern example in *examples/patterns/negatable-boolean-options.ts* now uses `negatableFlag()`, and *CHANGES.md* records the change.

Closes #801.

### CODE REVIEW

**Comment by dahlia** [comment]
@codex review

File: docs/cookbook.md [hunk 1]
```diff
@@ -925,37 +948,46 @@ const result = run(configParser);
...
 myapp
 
-# Disable colors and syntax, enable line numbers
+# Disable colors and syntax, enable line numbers explicitly
 myapp --no-colors --no-syntax --line-numbers
 
-# Disable code fence only
-myapp --no-code-fence
+# Enable colors explicitly when auto-detection would disable them
+myapp --colors
```
[the remark this answers is not in the document]
Comment (dahlia) [reply docs/cookbook.md:989]: Addressed in 52f253b7: clarified that auto is only the displayed default label and that the fallback value remains Boolean.

File: packages/core/src/primitives.test.ts [hunk 2]
```diff
@@ -1668,6 +1671,639 @@ describe("flag", () => {
...
+      positive: "--color",
+      negative: "--no-color",
+    });
+
+    const result = parseSync(parser, ["--no-color"]);
+
+    assert.ok(result.success);
+    if (result.success) {
+      assert.equal(result.value, false);
+    }
```
[the remark this answers is not in the document]
Comment (dahlia) [reply packages/core/src/primitives.test.ts:1700]: Addressed in 52f253b7: replaced the new Boolean equality assertions with assert.ok patterns.

File: packages/core/src/primitives.ts [hunk 3]
```diff
@@ -1862,6 +1862,457 @@ export function flag(
...
+ * @param names The positive and negative option names to parse.
+ * @param options Optional metadata and error customization.
+ * @returns A {@link Parser} that produces `true` for positive names and
+ *          `false` for negative names.
+ * @since 1.1.0
+ */
+export function negatableFlag(
+  names: NegatableFlagNames,
+  options: NegatableFlagOptions = {},
+): Parser<"sync", boolean, NegatableFlagState | undefined> {
```
[the remark this answers is not in the document]
Comment (dahlia) [reply packages/core/src/primitives.ts:2099]: Addressed in 52f253b7: documented the constructor-time TypeError cases on negatableFlag().

**Comment by dahlia** [comment]
@codex review

File: packages/core/src/primitives.ts [hunk 4]
```diff
@@ -1862,6 +1862,459 @@ export function flag(
...
+    },
+    suggest(_context, prefix) {
+      if (isSuggestionHidden(options.hidden)) {
+        return [];
+      }
+      const suggestions: Suggestion[] = [];
+      if (
+        prefix.startsWith("--") || prefix.startsWith("-") ||
+        prefix.startsWith("/")
+      ) {
```
[the remark this answers is not in the document]
Comment (dahlia) [reply packages/core/src/primitives.ts:2262]: Addressed in 99f456da: plus-prefixed negatable flag names are now included in option-name suggestions, with a regression test.

**Comment by dahlia** [comment]
@codex review

File: packages/core/src/primitives.ts [hunk 5]
```diff
@@ -1862,6 +1862,459 @@ export function flag(
...
+            : message`Required flag ${eOptionNames(optionNames)} is missing.`,
+        };
+      }
+      if (state.success) {
+        return { success: true, value: state.value };
+      }
+      return {
+        success: false,
+        error: message`${eOptionNames(optionNames)}: ${state.error}`,
+      };
```
[the remark this answers is not in the document]
Comment (dahlia) [reply packages/core/src/primitives.ts]: Addressed in be0b0882: completion now returns directly from the narrowed state shape.

File: packages/core/src/primitives.ts [hunk 6]
```diff
@@ -1862,6 +1862,459 @@ export function flag(
...
+      error: context.state.value === value
+        ? formatNegatableFlagDuplicateError(options, token)
+        : formatNegatableFlagConflictError(options, context.state.token, token),
+    };
+  }
+  return {
+    success: true,
+    next: {
+      ...context,
+      state: { success: true, value, token },
```
[the remark this answers is not in the document]
Comment (dahlia) [reply packages/core/src/primitives.ts]: Addressed in be0b0882: the negatable flag parser state now stores only the matched value and token.

**Comment by dahlia** [comment]
@codex review

File: packages/core/src/primitives.ts [hunk 7]
```diff
@@ -1862,6 +1862,459 @@ export function flag(
...
+): ParserResult<NegatableFlagState> {
+  if (context.state?.success) {
+    return {
+      success: false,
+      consumed: consumedOnFailure,
+      error: context.state.value === value
+        ? formatNegatableFlagDuplicateError(options, token)
+        : formatNegatableFlagConflictError(options, context.state.token, token),
+    };
+  }
```
[the remark this answers is not in the document]
Comment (dahlia) [reply packages/core/src/primitives.ts]: Addressed in be0b0882: the repeated-flag check now uses a nullish state check.

File: packages/core/src/primitives.ts [hunk 8]
```diff
@@ -1862,6 +1862,459 @@ export function flag(
...
+export type NegatableFlagState =
+  | {
+    readonly success: true;
+    readonly value: boolean;
+    readonly token: string;
+  }
+  | {
+    readonly success: false;
+    readonly error: Message;
+  };
```
[the remark this answers is not in the document]
Comment (dahlia) [reply packages/core/src/primitives.ts]: Addressed in be0b0882: NegatableFlagState now stores only the matched value and token.

File: packages/core/src/primitives.ts [hunk 9]
```diff
@@ -1862,6 +1862,447 @@ export function flag(
...
+    | Message
+    | ((optionName: string, value: string) => Message);
+
+  /**
+   * Custom error message when no matching flag is found.
+   */
+  noMatch?:
+    | Message
+    | ((invalidOption: string, suggestions: readonly string[]) => Message);
+}
```
[the remark this answers is not in the document]
Comment (dahlia) [reply packages/core/src/primitives.ts:1967]: Addressed in c3b951dd: all NegatableFlagErrorOptions fields are now readonly.

**Comment by dahlia** [comment]
@codex review

File: packages/core/src/primitives.ts [hunk 10]
```diff
@@ -1862,6 +1862,447 @@ export function flag(
...
+      return `negatableFlag({ ${args.join(", ")} })`;
+    },
+  };
+  Object.defineProperty(result, "placeholder", {
+    value: false,
+    configurable: true,
+    enumerable: false,
+    writable: false,
+  });
+  return result;
```
[the remark this answers is not in the document]
Comment (dahlia) [reply packages/core/src/primitives.ts:2322]: Addressed in bb5d3695: negatableFlag now validates fallback values as booleans and rejects non-boolean inputs.

File: packages/core/src/primitives.ts [hunk 11]
```diff
@@ -1862,6 +1862,447 @@ export function flag(
...
+        );
+      }
+
+      const prefixes = optionNames
+        .filter((name) =>
+          name.startsWith("--") ||
+          name.startsWith("/") ||
+          (name.startsWith("-") && name.length > 2)
+        )
+        .map((name) => name.startsWith("/") ? `${name}:` : `${name}=`);
```
[the remark this answers is not in the document]
Comment (dahlia) [reply packages/core/src/primitives.ts]: Addressed in bb5d3695: joined-value detection now computes prefixes lazily inside the option-name loop.

File: packages/core/src/primitives.ts [hunk 12]
```diff
@@ -1862,6 +1862,447 @@ export function flag(
...
+            error: formatNegatableFlagUnexpectedValueError(
+              options,
+              prefix.slice(0, -1),
+              value,
+            ),
+          };
+        }
+      }
+
+      const shortOptions = optionNames.filter((name) => name.match(/^-[^-]$/));
```
[the remark this answers is not in the document]
Comment (dahlia) [reply packages/core/src/primitives.ts]: Addressed in bb5d3695: bundled short-flag detection now scans option names directly without allocating a filtered array.

**Comment by dahlia** [comment]
@codex review

File: packages/core/src/primitives.ts [hunk 13]
```diff
@@ -1862,6 +1862,465 @@ export function flag(
...
+          options,
+        );
+      }
+
+      for (const name of optionNames) {
+        if (
+          !name.startsWith("--") &&
+          !name.startsWith("/") &&
+          !(name.startsWith("-") && name.length > 2)
+        ) {
```
[the remark this answers is not in the document]
Comment (dahlia) [reply packages/core/src/primitives.ts:2175]: Addressed in 071d1f58: plus-prefixed negatable flags now take the same joined-value rejection path as long and slash-prefixed names.

**Comment by dahlia** [comment]
@codex review
